Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs

```diff
@@ -271,21 +277,34 @@ public Task<BoolResult> UnregisterAsync(Context context, IReadOnlyList<ContentHa
 
             var operationContext = new OperationContext(context);
 
+            // Ideally, we want to remove content we know won't be used again for quite a while. We don't have that
+            // information, so we instead have an evictability metric. Here we obtain and sort by that evictability
+            // metric.
             var pageSize = _configuration.EvictionWindowSize;
+            var poolMultiplier = _configuration.MaximumEvictionPoolMultiplier;
+
+            float? takeoutFraction = null;
+            if (contentHashesWithInfo.Count > 0)
+            {
+                // We want to remove roughly the size of the pool divided by the amount of content at each iteration
+                takeoutFraction = 1 / contentHashesWithInfo.Count;

```

Comment:
This looks like integer division, in which case takeoutFraction = 0 always. To get around this problem, use 1.0 instead of 1.